File: web/admin/src/enums /ProxyTypeEnum.js
import Enum from "./Enum";


export default class ProxyTypeEnum extends Enum {
    static HTTP = new ProxyTypeEnum('HTTP', 'HTTP');
    static SOCKET5 = new ProxyTypeEnum('SOCKET5', 'SOCKET5');
    static SOCKET4 = new ProxyTypeEnum('SOCKET4', 'SOCKET4');

}

File: server/src/main/java/com/codeages/termiusplus/biz/server/enums/ProxyTypeEnum.java
package com.codeages.termiusplus.biz.server.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.net.Proxy;

@Getter
@AllArgsConstructor
public enum ProxyTypeEnum {
    HTTP(3L, Proxy.Type.HTTP),
    SOCKET4(1L, Proxy.Type.HTTP),
    SOCKET5(2L, Proxy.Type.SOCKS),
    ;

    private final Long code;

    private final Proxy.Type Type;
}
